app/assets/javascripts/discourse/components/composer-editor.js.es6

```diff
@@ -218,6 +223,17 @@ export default Ember.Component.extend({
     }
   },
 
+  _setUploadPosition(data) {
+    if (data.originalFiles.length > 1) {
+      let position = data.originalFiles.indexOf(data.files[0]) + 1;
```

Comment:
This works when you drag & drop multiple files (or select multiple files in the browse modal). How does it handle dragging & dropping **large** files one at a time?

Reply:
I really really love the idea of using filename, you would still translate it with a placeholder for filename but it would read way better:

`uploading: test.png` is so much clearer... just make sure that you handle the crazy case where people upload 2 files called "test.png" in one go (by selecting from diff folders for example, not 100% sure you can swing it that easily, but we don't want to break here for that)

also in some cases, cut and paste for example you have no proper filename to use, so maybe use `uploading: clipboard` which reads nice.